torch/_inductor/codegen/cuda/cutlass_lib_extensions/evt_extensions.py

```diff
@@ -41,13 +42,80 @@
     from cutlass.backend.evt.ir.tensor import (  # type: ignore[import-untyped, import-not-found]
         Tensor as CutlassTensor,
     )
-    from cutlass_library import DataType, EpilogueScheduleType, TileDescription
+    from cutlass_library import (
+        DataType,
+        EpilogueScheduleType,
+        LayoutType,
+        TileDescription,
+    )
 
+    import torch
     from torch._inductor.codegen.cuda import cuda_env
     from torch._inductor.utils import IndentedBuffer
 
     _CUTLASS_C_DTYPES = OrderedSet(dtype2ctype.values())  # type: ignore[var-annotated]
 
+    TORCH_TO_CUTLASS_DTYPE = {
+        torch.float32: DataType.f32,
+        torch.float16: DataType.f16,
+        torch.bfloat16: DataType.bf16,
+    }
+
+    def _check_contiguous(shape: tuple[int, ...], stride: tuple[int, ...]) -> bool:
+        if len(shape) != len(stride):
+            return False
+
+        running_product = 1
+        for shape_i, stride_i in zip(shape, stride):
+            if shape_i == 1:
+                continue
+            if stride_i != running_product:
+                return False
+            running_product *= shape_i
+
+        return True
+
+    def create_example_tensors(
+        read_names: list[str],
+        write_names: list[str],
+        buffer_renames: dict[str, str],
+        name_to_buffer: dict[str, Union[ComputedBuffer, InputBuffer]],
+    ):
+        example_tensors = {}
+
+        def cutlass_tensor_from_buffer(buffer: Union[ComputedBuffer, InputBuffer]):
+            shape = tuple(int(x) for x in buffer.get_layout().size)
+            stride = tuple(int(x) for x in buffer.get_layout().stride)
+
+            is_column_major = _check_contiguous(shape, stride)
+            is_row_major = _check_contiguous(shape[::-1], stride[::-1])
```

Comment:
int(symint) will throw e.g. if the symint is unbacked. We should be able to check this without converting the symints -> int. 

`buffer.get_layout().is_transposed` and `buffer.get_layout().is_contiguous`

Reply:
I do need to know whether row or column major though unfortunately (another suggestion for NVIDIA)

Reply:
for the symint thing, I also need them to be ints though for cutlass